tools/test/audio/comp_run.sh

```diff
@@ -5,14 +5,111 @@
 # stop on most errors
 set -e
 
-COMP=$1
-DIRECTION=$2
-BITS_IN=$3
-BITS_OUT=$4
-FS1=$5
-FS2=$6
-FN_IN=$7
-FN_OUT=$8
+usage ()
+{
+    cat <<EOFHELP
+Usage:     $0 <options> <comp direction bits_in bits_out fs_in fs_out input output>
+Example 1: $0 volume playback 16 16 48000 48000 input.raw output.raw
+Example 2: $0 -e volume_trace.txt -t volume_config.sh
+
+Where volume_config.sh could be e.g. next. Minimal configuration need is only
+the COMP line.
+
+# Volume component configuration
+COMP=volume
+DIRECTION=playback
+BITS_IN=16
+BITS_OUT=16
+CHANNELS_IN=2
+CHANNELS_OUT=2
+FS_IN=48000
+FS_OUT=48000
+FN_IN=input.raw
+FN_OUT=output.raw
+FN_TRACE:=trace.txt # This is default value if FN_TRACE is not set via -e option
+VALGRIND=yes
+EOFHELP
+}
+
+parse_args ()
+{
+    # Defaults
+    DIRECTION=playback
+    BITS_IN=16
+    BITS_OUT=
+    CHANNELS_IN=2
+    CHANNELS_OUT=
+    FS_IN=48000
+    FS_OUT=
+    VALGRIND=true
+    DEBUG=
+    SOURCED_CONFIG=false
+    FN_TRACE=
+    EXTRA_OPTS=
+
+    while getopts ":he:t:" opt; do
+	case "${opt}" in
+	    e)
+		FN_TRACE="${OPTARG}"
+		;;
+	    h)
+		usage
+		exit
+		;;
+	    t)
+		source "${OPTARG}"
+		SOURCED_CONFIG=true
+		;;
+	    *)
+		usage
+		exit 1
+		;;
+	esac
+    done
+
+    shift $((OPTIND -1))
+
+    if ! "$SOURCED_CONFIG"; then
+	[ $# -eq 8 ] || {
+	    usage "$0"
+	    exit 1
+	}
+
+	COMP=$1
+	DIRECTION=$2
+	BITS_IN=$3
+	BITS_OUT=$4
+	FS_IN=$5
+	FS_OUT=$6
+	FN_IN=$7
+	FN_OUT=$8
```

Comment:
shellcheck warnings?

Reply:
Yes, shellcheck thinks I should use `$VALGRIND_CMD "$CMD"`. But in that case valgrind things the whole testbench command is the binary to run. Do you ideas how to do this fix in way that still works?

Reply:
> Yes, shellcheck thinks I should use $VALGRIND_CMD "$CMD". 

"$CMD" is the tricky one, I wasn't referring to this one. I was only commenting about the easy ones like: `FN_OUT="$8"`.

There are two ways to deal with the `$CMD` quoting problem:
- Add a `# shellcheck disable=SC1234` silencer comment that states "thank you shellcheck but I know what I'm doing"
- Replace $CMD with a bash array `${CMD[@]}` https://mywiki.wooledge.org/BashGuide/Arrays
- Not do anything and not achieve shellcheck perfection. The fewer warnings the better, does not have to be zero.

Reply:
Shellcheck didn't complain about likes of FN_OUT=$8. OK, I'll fix those.